packages/coin-xlm/README.md

```diff
@@ -11,74 +12,103 @@ npm i @coolwallet/xlm
 
-### signTransaction
+const param = new TransactionBuilder(fundingAccount, {
+	fee: parseInt(moveDecimal(fee, 7), 10),
+	networkPassphrase: data.networkPassphrase,
+})
+	.addOperation(operation)
+	.setTimeout(TimeoutInfinite)
+	.addMemo(memo);
+
+const tx = param.build();
+
+const txData = {
+	from: publicKey,
+	to: publicKey,
+	amount: value * 10000000,
+	fee: tx.fee,
+	sequence: tx.sequence,
+	minTime: tx.timeBounds.minTime,
+	maxTime: tx.timeBounds.maxTime,
+	memoType: memoType,
+	memo: "",
+	isCreate: true
+}
+
+const signTxData = {
+    transport,
+    appPrivateKey,
+    appId,
+    transaction,
+    protocol,
+    addressIndex
+}
+
+const signatureTx = await xlm.signTransaction(signTxData)
+```
+In construct, you can choose the chain you want to implement.
 
-We expect you to build transactions with the [official stellar sdk](https://github.com/stellar/js-stellar-sdk). You can easily use the `.signatureBase()` method to get the buffer that CoolWallet needs for signing.
 
-```javascript
-import * as Stellar from 'stellar-sdk';
-const passphrase = 'Public Global Stellar Network ; September 2015';
 
-const server = new Stellar.Server('https://horizon.stellar.org')
+## Methods
 
-const fundingAccount = await server.loadAccount(account)
-const operation = Stellar.Operation.payment({
-  destination: 'GBE6DJHSIR6RLPCTJLIYBCA7VUOFNJ5YW6MSAOJL3QQ4E2BI3OA5EFP4',
-  asset: Stellar.Asset.native(),
-  amount: '1',
-})
+### getAddress
 
-let txBuilder = new Stellar.TransactionBuilder(fundingAccount, {
-  fee: 100,
-  networkPassphrase: passphrase,
-})
-  .addOperation(operation)
-  .setTimeout(Stellar.TimeoutInfinite);
+#### Description
+
+CoolWallet currently support 2 derivation path, the default one is **SLIP0010**.
 
-let tx = txBuilder.build();
-const signatureBase = tx.signatureBase(); // this is what we need.
+```none
+m/44'/94'/0'
+```
 
-// Sign with CoolWallet
-const signature = await XLM.signTransaction(signatureBase, accountIndex)
+We call the fourth parameter **protocol**, which can only be either `'BIP44'` or `'SLIP0010'`. You will need to specify the protocol parameter when you're signing a transaction.
 
+```javascript
+async getAddress(
+    transport: Transport, 
+    appPrivateKey: string, 
+    appId: string, 
+    addressIndex: number
+    ): Promise<string> 
 ```
 
-### Build transaction object
+#### Arguments
+|      Arg      |                  Description                 |    Type   | Required |
+|:-------------:|:--------------------------------------------:|:---------:|:--------:|
+|   transport   | Object to communicate with CoolWallet device | Transport |   TRUE   |
+| appPrivateKey |   Private key for the connected application  |   string  |   TRUE   |
+|     appId     |       ID for the connected application       |   string  |   TRUE   |
+|    protocol   |      Define the protocol you're signing      |  PROTOCOL |   FALSE  |
```

Comment:
the type of `protocol` shall be `string` unless we explicitly declare the type `PROTOCOL`

Reply:
oh I see it's actually `PROTOCOL` type, so I think in our usage example we shall use `PROTOCOL.BIP44` to avoid confusion.